test/index.coffee

```diff
@@ -0,0 +1,84 @@
+Promise = require 'bluebird'
+{ expect } = require 'chai'
+_ = require 'lodash'
+Docker = require 'dockerode'
+{ DockerProgress } = require 'docker-progress'
+
+gc = require('../lib/index')
+
+SKIP_GC_TEST = process.env.SKIP_GC_TEST? || false
+DOCKER_PATH = process.env.DOCKER_PATH || '/var/run/docker.sock'
+docker = new Docker({ socketPath: DOCKER_PATH, Promise })
+dockerProgress = new DockerProgress({ socketPath: DOCKER_PATH })
+
+promiseToBool = (p) ->
+	p.then(-> return true).catchReturn(false)
+
+describe 'Garbage collection', () ->
+	# This test case is a little weird, it requires that no other images are present on
+	# the system to ensure that the correct one is being removed. Because of this, you
+	# can use the SKIP_GC_TEST env var to inform the test suite not to run this test
+	it 'should remove the LRU image', () ->
+		this.timeout(600000)
+		return Promise.resolve() if SKIP_GC_TEST
+		# first pull some images, so we know in which order they are referenced
+
+		# Create the promises with a standard map so that they're done in the correct
+		# order (bluebird doesn't guarantee this)
+		pullPromises = [
```

Comment:
We can't do this because of the comment above, bluebird won't run them in the correct order, which is essential for this test.